File: src/python/code_1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, n in enumerate(nums):
            if abs(target - n) in seen:
                return [seen[abs(target - n)], i]
            else:
                seen[n] = i

File: java/src/main/java/leetcube/phaserush/Code1.java
package leetcube.phaserush;

import java.util.HashMap;
import java.util.Map;

public final class Code1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int curr = nums[i];
            int currTarget = target - curr;
            if (seen.containsKey(currTarget)) {
                return new int[]{i, seen.get(currTarget)};
            }
            seen.put(curr, i);
        }
        return null;
    }
}